content/tutorials/first-aws-environment.md

```diff
@@ -0,0 +1,181 @@
+# Geodesic is the base image for our tutorials image and it isn't tiny by any means
+# Go grab a coffee while you wait ☕️
+make build
+
+# Run our docker image
+make run
+```
+
+This will build the `tutorials` image locally on your machine and then runs it while also mounting the various tutorial folders so you can edit them on your host machine or in the container and the changes will propogate either direction.
+
+![Tutorial Shell](/assets/tutorials-3-tutorials-shell.png)
+
+Now that we're running inside of our container, let's start a new shell as your AWS profile and get into our specific tutorial directory:
+
+```bash
+# First let's enter a new shell with credentials from your AWS profile
+# This will enable our interactions with AWS through Terraform in the rest of the tutorial to be properly authenticated
+aws-vault exec $YOUR_PROFILE -- bash
+
+cd /tutorials/03-first-aws-environment
+```
+
+This `03-first-aws-environment/` directory that you're now in should be looking a little bit familiar from our first tutorial:
+
+* We've got a `components/` directory which holds terraform components that we're looking to `apply` to our AWS Account.
+* We've got a `stacks/` directory which holds the declarative descriptions of 3 environments:
+  * `uw2-dev.yaml`: Our dev environment hosted in us-west-2.
+  * `uw2-prod.yaml`: Our prod environment hosted in the same region.
+  * `ue2-root.yaml`: Our global resources hosted in us-east-2.
+* We've also got a `bin/uniquify.sh` file that we'll discuss shortly.
+
+Cool, let's jump into our next step and actually do something with this small project.
+
+### 2. Build and generate our `tfstate-backend` for this project
+
+In our last tutorial, we showed `atmos` being used to apply some fairly simple terraform projects that didn't actually create any resources in the cloud. For this tutorial however, we're going to be creating a simple static site in AWS and as part of that we should be sure to follow one of the most important best practices in terraform: **Use remote state**. Luckily, `atmos` and stacks make that easy on us by providing a simple abstraction layer to configure all our terraform components with a single backend config. In this example we'll be using the S3 backend and [the tfstate-backend component](https://github.com/cloudposse/terraform-aws-components/tree/master/modules/tfstate-backend) to automate the resources we need for that backend, so let's dive into that:
+
+1. To start off, we need to do something funky that you wouldn't need to do in a normal project setting: We need to create unique names for our project since we're dealing with S3 buckets and their names are global across all AWS accounts. To make that easy on you, we've included `bin/uniquify.sh` which was mentioned previously. This small script generates some random strings and utilizes [`yq`](https://mikefarah.gitbook.io/yq/commands/evaluate) to update some of our stack files to include random names. Let's run that now:
```

Comment:
```suggestion
1. To start off, we need to do something funky to generate unique resource names. This is because this tutorial used by many people, so it's not something a normal project setting would need. That's because this tutorial is creating S3 buckets and their names need to be globally unique across all AWS accounts. To make that easy on you, we've included a little helper script (`bin/uniquify.sh`) which was mentioned previously. This small script generates some random strings and utilizes [`yq`](https://mikefarah.gitbook.io/yq/commands/evaluate) to update some of our stack files to include random names. Let's run that now:
```